ProjectManager/ProjectManager/View/Main/MainViewController.swift

```diff
@@ -0,0 +1,205 @@
+    private lazy var  doingTableView = listTableView()
+    private lazy var  doneTableView = listTableView()
+    
+    private let viewModel: MainViewModelInOut
+    private let container: Container
+    private var disposebag = DisposeBag()
+    
+    init(viewModel: MainViewModelInOut, container: Container) {
+        self.viewModel = viewModel
+        self.container = container
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setInitailView()
+    }
+    
+    private func setInitailView() {
+        self.view.backgroundColor = .systemGray5
+        self.view.addSubview(mainStackView)
+        mainStackView.snp.makeConstraints {
+            $0.edges.equalTo(self.view.safeAreaLayoutGuide)
+        }
+        setTableView()
+        setNavigationBar()
+    }
+    
+    private func setNavigationBar() {
+        self.title = "Project Manger"
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(didTapAddButton))
+    }
+    
+    @objc private func didTapAddButton() {
+        self.present(container.makeAddViewController(), animated: true)
+    }
+    
+    private func didtapCell(_ listItem: ListItem) {
+        self.present(container.makeEditViewController(listItem), animated: true)
+    }
+    
+    private func setTableView() {
+        bindTableView(todoTableView,
+                      list: viewModel.todoList,
+                      headerView: todoHeaderView,
+                      type: .todo)
+        bindTableView(doingTableView,
+                      list: viewModel.doingList,
+                      headerView: doingHeaderView,
+                      type: .doing)
+        bindTableView(doneTableView,
+                      list: viewModel.doneList,
+                      headerView: doneHeaderView,
+                      type: .done)
+    }
+    
+    private func bindTableView(_ tableView: UITableView, list: Driver<[ListItem]>, headerView: HeaderView, type: ListType) {
+        list
+            .drive(tableView.rx.items(cellIdentifier: "\(ListTableViewCell.self)", cellType: ListTableViewCell.self)) { [weak self] index, item, cell in
+                guard let self = self else {
+                    return
+                }
+                
+                cell.setViewContents(item, isOver: self.viewModel.isOverDeadline(listItem: item))
+            }
+            .disposed(by: disposebag)
+        
+        list
+          .map { "\($0.count)"}
+          .drive(headerView.countLabel.rx.text)
+          .disposed(by: disposebag)
+        
+        tableView.rx.itemSelected
+            .bind(onNext: { index in
+                
+                self.viewModel.peekList(index: index.row, type: type) {
+                    self.didtapCell($0)
+                }
```

Comment:
흠.. 뷰모델로 input을 전달하는 흐름과 뷰모델에서 output 을 받아 뷰를 업데이트하는 흐름이 명확하게 나눠지지 못한 것 같습니다.
```suggestion
                self.viewModel.peekList(index: index.row, type: type)
```
여기선 input 전달까지만 수행하고, 다른 곳에서 뷰모델의 output을 받아 뷰를 업데이트하는 로직을 구현해야 합니다.
output의 전달은 함수 내부에서 로직을 수행한 뒤 지금처럼 함수 인자로 클로저를 전달하는 게 아니라, 뷰모델에 output 전달 통로를 정의하고 그곳을 통해 결과값을 전달하는 형태가 되어야 할 것 같아요.

`MainViewModel`에 `ListItem`을 방출하는 `presentEditView` output을 정의하고, 뷰모델에서 `peekList`가 호출됐을 때 로직을 수행한 뒤 `presentEditView`를 통해 `ListItem`을 방출하면 뷰에선 `presentEditView`가 방출한 값을 받아 `EditViewController`를 띄워주도록요.

Reply:
(`peek`은 데이터 구조에서 선택 보단 다른 의미로 쓰이는 경우가 많아 select 와 같은 명확한 이름으로 변경하는 게 좋을 것 같아요 +_+)

Reply:
이 부분에 대해서 고민을 해봤을 때 굳이 item을 직접 넘겨주기 보다 몇 번째 셀이 선택됐는지, 그 셀은 어떤 타입의 item인지만 editViewModel로 전달해줘서 그 곳에서 받은 정보로 storage에서 item을 뽑아 오는게 좋을 것 같다는 생각이 들어 말씀드린 방식으로 수정하였습니다